File: hackertracker/src/main/java/com/shortstack/hackertracker/Vendors/VendorsFragment.java
package com.shortstack.hackertracker.Vendors;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.view.InflateException;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import com.shortstack.hackertracker.Activity.HomeActivity;
import com.shortstack.hackertracker.Adapter.VendorAdapter;
import com.shortstack.hackertracker.Common.Constants;
import com.shortstack.hackertracker.Fragment.HackerTrackerFragment;
import com.shortstack.hackertracker.Model.Default;
import com.shortstack.hackertracker.Model.Vendor;
import com.shortstack.hackertracker.R;

import java.util.List;

/**
 * Created by Whitney Champion on 3/29/14.
 */
public class VendorsFragment extends HackerTrackerFragment {

    public Vendor[] vendorData;
    public VendorAdapter adapter;
    private ListView list;
    private static View rootView;
    private Context context;
    private static final String ARG_SECTION_NUMBER = "section_number";

    public static VendorsFragment newInstance(int sectionNumber) {
        VendorsFragment fragment = new VendorsFragment();
        Bundle args = new Bundle();
        args.putInt(ARG_SECTION_NUMBER, sectionNumber);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        context = inflater.getContext();

        if (rootView != null) {
            ViewGroup parent = (ViewGroup) rootView.getParent();
            if (parent != null)
                parent.removeView(rootView);
        }
        try {
            rootView = inflater.inflate(R.layout.fragment_vendors, container, false);
        } catch (InflateException e) {
        }

        list = (ListView) rootView.findViewById(R.id.list_vendors);

        String date = "1";

        // get vendors
        List<Default> vendors = getItemByDate(date, Constants.TYPE_VENDOR);
        if (vendors.size() > 0) {

            vendorData = vendors.toArray(new Vendor[vendors.size()]);

            adapter = new VendorAdapter(context, R.layout.row_vendor, vendorData);

            list.setAdapter(adapter);

        }

        return rootView;
    }

}

File: hackertracker/src/main/java/com/shortstack/hackertracker/vendors/VendorsFragment.kt
package com.shortstack.hackertracker.vendors

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.pedrogomez.renderers.RendererAdapter
import com.pedrogomez.renderers.RendererBuilder
import com.shortstack.hackertracker.Application.App
import com.shortstack.hackertracker.Model.Vendor
import com.shortstack.hackertracker.R
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.fragment_recyclerview.*


class VendorsFragment : Fragment() {

    override fun onCreateView(inflater : LayoutInflater?, container : ViewGroup?, savedInstanceState : Bundle?) : View? {
        return inflater!!.inflate(R.layout.fragment_recyclerview, container, false)
    }

    override fun onViewCreated(view : View?, savedInstanceState : Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val layout = LinearLayoutManager(context)
        list.layoutManager = layout

        val rendererBuilder = RendererBuilder<Any>()
                .bind(Vendor::class.java, VendorRenderer())

        val adapter = RendererAdapter<Any>(rendererBuilder)
        list.adapter = adapter

        getVendors()
    }

    private fun getVendors() {
        setProgressIndicator(true)

        App.application.databaseController.getVendors()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe({
                    setProgressIndicator(false)
                    showVendors(it.toTypedArray())
                }, {
                    if (isActive()) {
                        setProgressIndicator(false)
                        showLoadingVendorsError()
                    }

                })
    }


    private fun setProgressIndicator(active : Boolean) {
        loading_progress.visibility = if (active) View.VISIBLE else View.GONE
    }

    private fun showVendors(vendors : Array<Vendor>) {
        (list.adapter as RendererAdapter<*>).addAllAndNotify(vendors.toMutableList())
    }

    private fun showLoadingVendorsError() {
        Toast.makeText(context, "Could not load vendors.", Toast.LENGTH_SHORT).show()
    }

    private fun isActive() : Boolean {
        return isAdded
    }


    companion object {
        fun newInstance() : VendorsFragment {
            return VendorsFragment()
        }
    }
}



